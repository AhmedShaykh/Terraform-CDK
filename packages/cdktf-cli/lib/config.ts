import * as fs from 'fs-extra';
import * as path from 'path';
import { Language } from './get/base';

const CONFIG_FILE = 'cdktf.json'
const DEFAULTS = {
  output: 'cdktf.out',
  codeMakerOutput: '.gen'
}

function isPresent(input: string[] | undefined): boolean {
  return Array.isArray(input) && input.length > 0
}

export interface Config {
  readonly app?: string;
  readonly language?: Language;
  readonly output: string;
  readonly codeMakerOutput: string;
  readonly terraformProviders?: string[];
  readonly terraformModules?: string[];
  checkCodeMakerOutput?: boolean;
}

export function readConfigSync(): Config {
  const configFile = path.join(process.cwd(), CONFIG_FILE)
  let config: Config = DEFAULTS;
  if (fs.existsSync(configFile)) {
    config = {
      ...config,
      ...JSON.parse(fs.readFileSync(configFile).toString())
    };
  }

  config.checkCodeMakerOutput = isPresent(config.terraformModules) || isPresent(config.terraformProviders)

  return config;
}