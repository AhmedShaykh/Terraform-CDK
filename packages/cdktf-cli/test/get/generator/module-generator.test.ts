import * as fs from 'fs';
import * as os from 'os';
import * as path from 'path';
import { ConstructsMaker, Language } from '../../../lib/get/constructs-maker'
import { TerraformModuleConstraint  } from '../../../lib/config'
import { expectModuleToMatchSnapshot } from '../util';

test('generate some modules', async () => {
  jest.setTimeout(20000)

  const workdir = fs.mkdtempSync(path.join(os.tmpdir(), 'module-generator.test'));
  const constraint = new TerraformModuleConstraint('terraform-aws-modules/eks/aws@7.0.1')

  const maker = new ConstructsMaker({codeMakerOutput: workdir, targetLanguage: Language.TYPESCRIPT}, [constraint])
  await maker.generate()

  const output = fs.readFileSync(path.join(workdir, 'modules/terraform-aws-modules/eks/aws.ts'), 'utf-8');
  expect(output).toMatchSnapshot();
});

expectModuleToMatchSnapshot('no module outputs', 'generator', 'module-no-outputs.test.fixture.tf');