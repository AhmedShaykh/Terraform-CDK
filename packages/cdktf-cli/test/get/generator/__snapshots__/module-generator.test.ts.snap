// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate some modules 1`] = `
"// generated by cdktf get
// terraform-aws-modules/eks/aws
import { TerraformModule } from 'cdktf';
import { Construct } from 'constructs';
export interface TerraformAwsModulesEksAwsOptions {
  /**
   * Whether to attach the module managed cluster autoscaling iam policy to the default worker IAM role. This requires \`manage_worker_autoscaling_policy = true\`
   * @default true
   */
  readonly attachWorkerAutoscalingPolicy?: boolean;
  /**
   * Whether to attach the Amazon managed \`AmazonEKS_CNI_Policy\` IAM policy to the default worker IAM role. WARNING: If set \`false\` the permissions must be assigned to the \`aws-node\` DaemonSet pods via another method or nodes will not be able to join the cluster.
   * @default true
   */
  readonly attachWorkerCniPolicy?: boolean;
  /**
   * Whether to create a security group for the cluster or attach the cluster to \`cluster_security_group_id\`.
   * @default true
   */
  readonly clusterCreateSecurityGroup?: boolean;
  /**
   * Timeout value when creating the EKS cluster.
   * @default 15m
   */
  readonly clusterCreateTimeout?: string;
  /**
   * Timeout value when deleting the EKS cluster.
   * @default 15m
   */
  readonly clusterDeleteTimeout?: string;
  /**
   * A list of the desired control plane logging to enable. For more information, see Amazon EKS Control Plane Logging documentation (https://docs.aws.amazon.com/eks/latest/userguide/control-plane-logs.html)
   * @default 
   */
  readonly clusterEnabledLogTypes?: string[];
  /**
   * Indicates whether or not the Amazon EKS private API server endpoint is enabled.
   */
  readonly clusterEndpointPrivateAccess?: boolean;
  /**
   * Indicates whether or not the Amazon EKS public API server endpoint is enabled.
   * @default true
   */
  readonly clusterEndpointPublicAccess?: boolean;
  /**
   * IAM role name for the cluster. Only applicable if manage_cluster_iam_resources is set to false.
   */
  readonly clusterIamRoleName?: string;
  /**
   * If a KMS Key ARN is set, this key will be used to encrypt the corresponding log group. Please be sure that the KMS Key has an appropriate key policy (https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/encrypt-log-data-kms.html)
   */
  readonly clusterLogKmsKeyId?: string;
  /**
   * Number of days to retain log events. Default retention - 90 days.
   * @default 90
   */
  readonly clusterLogRetentionInDays?: number;
  /**
   * Name of the EKS cluster. Also used as a prefix in names of related resources.
   */
  readonly clusterName: string;
  /**
   * If provided, the EKS cluster will be attached to this security group. If not given, a security group will be created with necessary ingress/egress to work with the workers
   */
  readonly clusterSecurityGroupId?: string;
  /**
   * Kubernetes version to use for the EKS cluster.
   * @default 1.14
   */
  readonly clusterVersion?: string;
  /**
   * Where to save the Kubectl config file (if \`write_kubeconfig = true\`). Assumed to be a directory if the value ends with a forward slash \`/\`.
   * @default ./
   */
  readonly configOutputPath?: string;
  /**
   * If provided, all IAM roles will be created on this path.
   * @default /
   */
  readonly iamPath?: string;
  /**
   * Any additional arguments to pass to the authenticator such as the role to assume. e.g. [\\"-r\\", \\"MyEksRole\\"].
   * @default 
   */
  readonly kubeconfigAwsAuthenticatorAdditionalArgs?: string[];
  /**
   * Command to use to fetch AWS EKS credentials.
   * @default aws-iam-authenticator
   */
  readonly kubeconfigAwsAuthenticatorCommand?: string;
  /**
   * Default arguments passed to the authenticator command. Defaults to [token -i $cluster_name].
   * @default 
   */
  readonly kubeconfigAwsAuthenticatorCommandArgs?: string[];
  /**
   * Environment variables that should be used when executing the authenticator. e.g. { AWS_PROFILE = \\"eks\\"}.
   * @default [object Object]
   */
  readonly kubeconfigAwsAuthenticatorEnvVariables?: { [key: string]: string };
  /**
   * Override the default name used for items kubeconfig.
   */
  readonly kubeconfigName?: string;
  /**
   * Command to run for local-exec resources. Must be a shell-style interpreter. If you are on Windows Git Bash is a good choice.
   * @default /bin/sh,-c
   */
  readonly localExecInterpreter?: string[];
  /**
   * Whether to apply the aws-auth configmap file.
   * @default true
   */
  readonly manageAwsAuth?: boolean;
  /**
   * Whether to let the module manage cluster IAM resources. If set to false, cluster_iam_role_name must be specified.
   * @default true
   */
  readonly manageClusterIamResources?: boolean;
  /**
   * Whether to let the module manage the cluster autoscaling iam policy.
   * @default true
   */
  readonly manageWorkerAutoscalingPolicy?: boolean;
  /**
   * Whether to let the module manage worker IAM resources. If set to false, iam_instance_profile_name must be specified for workers.
   * @default true
   */
  readonly manageWorkerIamResources?: boolean;
  /**
   * Additional AWS account numbers to add to the aws-auth configmap. See examples/basic/variables.tf for example format.
   * @default 
   */
  readonly mapAccounts?: string[];
  /**
   * Additional IAM roles to add to the aws-auth configmap. See examples/basic/variables.tf for example format.
   * @default 
   */
  readonly mapRoles?: any;
  /**
   * Additional IAM users to add to the aws-auth configmap. See examples/basic/variables.tf for example format.
   * @default 
   */
  readonly mapUsers?: any;
  /**
   * If provided, all IAM roles will be created with this permissions boundary attached.
   */
  readonly permissionsBoundary?: string;
  /**
   * A list of subnets to place the EKS cluster and workers within.
   */
  readonly subnets: string[];
  /**
   * A map of tags to add to all resources.
   * @default [object Object]
   */
  readonly tags?: { [key: string]: string };
  /**
   * VPC where the cluster and workers will be deployed.
   */
  readonly vpcId: string;
  /**
   * A list of additional security group ids to attach to worker instances
   * @default 
   */
  readonly workerAdditionalSecurityGroupIds?: string[];
  /**
   * Name filter for AWS EKS worker AMI. If not provided, the latest official AMI for the specified 'cluster_version' is used.
   */
  readonly workerAmiNameFilter?: string;
  /**
   * Name filter for AWS EKS Windows worker AMI. If not provided, the latest official AMI for the specified 'cluster_version' is used.
   */
  readonly workerAmiNameFilterWindows?: string;
  /**
   * The ID of the owner for the AMI to use for the AWS EKS workers. Valid values are an AWS account ID, 'self' (the current account), or an AWS owner alias (e.g. 'amazon', 'aws-marketplace', 'microsoft').
   * @default 602401143452
   */
  readonly workerAmiOwnerId?: string;
  /**
   * The ID of the owner for the AMI to use for the AWS EKS Windows workers. Valid values are an AWS account ID, 'self' (the current account), or an AWS owner alias (e.g. 'amazon', 'aws-marketplace', 'microsoft').
   * @default 801119661308
   */
  readonly workerAmiOwnerIdWindows?: string;
  /**
   * Whether to create initial lifecycle hooks provided in worker groups.
   */
  readonly workerCreateInitialLifecycleHooks?: boolean;
  /**
   * Whether to create a security group for the workers or attach the workers to \`worker_security_group_id\`.
   * @default true
   */
  readonly workerCreateSecurityGroup?: boolean;
  /**
   * A list of maps defining worker group configurations to be defined using AWS Launch Configurations. See workers_group_defaults for valid keys.
   * @default 
   */
  readonly workerGroups?: any;
  /**
   * A list of maps defining worker group configurations to be defined using AWS Launch Templates. See workers_group_defaults for valid keys.
   * @default 
   */
  readonly workerGroupsLaunchTemplate?: any;
  /**
   * If provided, all workers will be attached to this security group. If not given, a security group will be created with necessary ingress/egress to work with the EKS cluster.
   */
  readonly workerSecurityGroupId?: string;
  /**
   * Minimum port number from which pods will accept communication. Must be changed to a lower value if some pods in your cluster will expose a port lower than 1025 (e.g. 22, 80, or 443).
   * @default 1025
   */
  readonly workerSgIngressFromPort?: number;
  /**
   * Additional policies to be added to workers
   * @default 
   */
  readonly workersAdditionalPolicies?: string[];
  /**
   * Override default values for target groups. See workers_group_defaults_defaults in local.tf for valid keys.
   * @default [object Object]
   */
  readonly workersGroupDefaults?: any;
  /**
   * User defined workers role name.
   */
  readonly workersRoleName?: string;
  /**
   * Whether to write the aws-auth configmap file.
   * @default true
   */
  readonly writeAwsAuthConfig?: boolean;
  /**
   * Whether to write a Kubectl config file containing the cluster configuration. Saved to \`config_output_path\`.
   * @default true
   */
  readonly writeKubeconfig?: boolean;
}
export class TerraformAwsModulesEksAws extends TerraformModule {
  private readonly inputs: { [name: string]: any } = { }
  public constructor(scope: Construct, id: string, options: TerraformAwsModulesEksAwsOptions) {
    super(scope, id, {
      source: 'terraform-aws-modules/eks/aws',
      version: '7.0.1',
    });
    this.attachWorkerAutoscalingPolicy = options.attachWorkerAutoscalingPolicy;
    this.attachWorkerCniPolicy = options.attachWorkerCniPolicy;
    this.clusterCreateSecurityGroup = options.clusterCreateSecurityGroup;
    this.clusterCreateTimeout = options.clusterCreateTimeout;
    this.clusterDeleteTimeout = options.clusterDeleteTimeout;
    this.clusterEnabledLogTypes = options.clusterEnabledLogTypes;
    this.clusterEndpointPrivateAccess = options.clusterEndpointPrivateAccess;
    this.clusterEndpointPublicAccess = options.clusterEndpointPublicAccess;
    this.clusterIamRoleName = options.clusterIamRoleName;
    this.clusterLogKmsKeyId = options.clusterLogKmsKeyId;
    this.clusterLogRetentionInDays = options.clusterLogRetentionInDays;
    this.clusterName = options.clusterName;
    this.clusterSecurityGroupId = options.clusterSecurityGroupId;
    this.clusterVersion = options.clusterVersion;
    this.configOutputPath = options.configOutputPath;
    this.iamPath = options.iamPath;
    this.kubeconfigAwsAuthenticatorAdditionalArgs = options.kubeconfigAwsAuthenticatorAdditionalArgs;
    this.kubeconfigAwsAuthenticatorCommand = options.kubeconfigAwsAuthenticatorCommand;
    this.kubeconfigAwsAuthenticatorCommandArgs = options.kubeconfigAwsAuthenticatorCommandArgs;
    this.kubeconfigAwsAuthenticatorEnvVariables = options.kubeconfigAwsAuthenticatorEnvVariables;
    this.kubeconfigName = options.kubeconfigName;
    this.localExecInterpreter = options.localExecInterpreter;
    this.manageAwsAuth = options.manageAwsAuth;
    this.manageClusterIamResources = options.manageClusterIamResources;
    this.manageWorkerAutoscalingPolicy = options.manageWorkerAutoscalingPolicy;
    this.manageWorkerIamResources = options.manageWorkerIamResources;
    this.mapAccounts = options.mapAccounts;
    this.mapRoles = options.mapRoles;
    this.mapUsers = options.mapUsers;
    this.permissionsBoundary = options.permissionsBoundary;
    this.subnets = options.subnets;
    this.tags = options.tags;
    this.vpcId = options.vpcId;
    this.workerAdditionalSecurityGroupIds = options.workerAdditionalSecurityGroupIds;
    this.workerAmiNameFilter = options.workerAmiNameFilter;
    this.workerAmiNameFilterWindows = options.workerAmiNameFilterWindows;
    this.workerAmiOwnerId = options.workerAmiOwnerId;
    this.workerAmiOwnerIdWindows = options.workerAmiOwnerIdWindows;
    this.workerCreateInitialLifecycleHooks = options.workerCreateInitialLifecycleHooks;
    this.workerCreateSecurityGroup = options.workerCreateSecurityGroup;
    this.workerGroups = options.workerGroups;
    this.workerGroupsLaunchTemplate = options.workerGroupsLaunchTemplate;
    this.workerSecurityGroupId = options.workerSecurityGroupId;
    this.workerSgIngressFromPort = options.workerSgIngressFromPort;
    this.workersAdditionalPolicies = options.workersAdditionalPolicies;
    this.workersGroupDefaults = options.workersGroupDefaults;
    this.workersRoleName = options.workersRoleName;
    this.writeAwsAuthConfig = options.writeAwsAuthConfig;
    this.writeKubeconfig = options.writeKubeconfig;
  }
  public get attachWorkerAutoscalingPolicy(): boolean | undefined {
    return this.inputs['attach_worker_autoscaling_policy'] as boolean | undefined;
  }
  public set attachWorkerAutoscalingPolicy(value: boolean | undefined) {
    this.inputs['attach_worker_autoscaling_policy'] = value;
  }
  public get attachWorkerCniPolicy(): boolean | undefined {
    return this.inputs['attach_worker_cni_policy'] as boolean | undefined;
  }
  public set attachWorkerCniPolicy(value: boolean | undefined) {
    this.inputs['attach_worker_cni_policy'] = value;
  }
  public get clusterCreateSecurityGroup(): boolean | undefined {
    return this.inputs['cluster_create_security_group'] as boolean | undefined;
  }
  public set clusterCreateSecurityGroup(value: boolean | undefined) {
    this.inputs['cluster_create_security_group'] = value;
  }
  public get clusterCreateTimeout(): string | undefined {
    return this.inputs['cluster_create_timeout'] as string | undefined;
  }
  public set clusterCreateTimeout(value: string | undefined) {
    this.inputs['cluster_create_timeout'] = value;
  }
  public get clusterDeleteTimeout(): string | undefined {
    return this.inputs['cluster_delete_timeout'] as string | undefined;
  }
  public set clusterDeleteTimeout(value: string | undefined) {
    this.inputs['cluster_delete_timeout'] = value;
  }
  public get clusterEnabledLogTypes(): string[] | undefined {
    return this.inputs['cluster_enabled_log_types'] as string[] | undefined;
  }
  public set clusterEnabledLogTypes(value: string[] | undefined) {
    this.inputs['cluster_enabled_log_types'] = value;
  }
  public get clusterEndpointPrivateAccess(): boolean | undefined {
    return this.inputs['cluster_endpoint_private_access'] as boolean | undefined;
  }
  public set clusterEndpointPrivateAccess(value: boolean | undefined) {
    this.inputs['cluster_endpoint_private_access'] = value;
  }
  public get clusterEndpointPublicAccess(): boolean | undefined {
    return this.inputs['cluster_endpoint_public_access'] as boolean | undefined;
  }
  public set clusterEndpointPublicAccess(value: boolean | undefined) {
    this.inputs['cluster_endpoint_public_access'] = value;
  }
  public get clusterIamRoleName(): string | undefined {
    return this.inputs['cluster_iam_role_name'] as string | undefined;
  }
  public set clusterIamRoleName(value: string | undefined) {
    this.inputs['cluster_iam_role_name'] = value;
  }
  public get clusterLogKmsKeyId(): string | undefined {
    return this.inputs['cluster_log_kms_key_id'] as string | undefined;
  }
  public set clusterLogKmsKeyId(value: string | undefined) {
    this.inputs['cluster_log_kms_key_id'] = value;
  }
  public get clusterLogRetentionInDays(): number | undefined {
    return this.inputs['cluster_log_retention_in_days'] as number | undefined;
  }
  public set clusterLogRetentionInDays(value: number | undefined) {
    this.inputs['cluster_log_retention_in_days'] = value;
  }
  public get clusterName(): string {
    return this.inputs['cluster_name'] as string;
  }
  public set clusterName(value: string) {
    this.inputs['cluster_name'] = value;
  }
  public get clusterSecurityGroupId(): string | undefined {
    return this.inputs['cluster_security_group_id'] as string | undefined;
  }
  public set clusterSecurityGroupId(value: string | undefined) {
    this.inputs['cluster_security_group_id'] = value;
  }
  public get clusterVersion(): string | undefined {
    return this.inputs['cluster_version'] as string | undefined;
  }
  public set clusterVersion(value: string | undefined) {
    this.inputs['cluster_version'] = value;
  }
  public get configOutputPath(): string | undefined {
    return this.inputs['config_output_path'] as string | undefined;
  }
  public set configOutputPath(value: string | undefined) {
    this.inputs['config_output_path'] = value;
  }
  public get iamPath(): string | undefined {
    return this.inputs['iam_path'] as string | undefined;
  }
  public set iamPath(value: string | undefined) {
    this.inputs['iam_path'] = value;
  }
  public get kubeconfigAwsAuthenticatorAdditionalArgs(): string[] | undefined {
    return this.inputs['kubeconfig_aws_authenticator_additional_args'] as string[] | undefined;
  }
  public set kubeconfigAwsAuthenticatorAdditionalArgs(value: string[] | undefined) {
    this.inputs['kubeconfig_aws_authenticator_additional_args'] = value;
  }
  public get kubeconfigAwsAuthenticatorCommand(): string | undefined {
    return this.inputs['kubeconfig_aws_authenticator_command'] as string | undefined;
  }
  public set kubeconfigAwsAuthenticatorCommand(value: string | undefined) {
    this.inputs['kubeconfig_aws_authenticator_command'] = value;
  }
  public get kubeconfigAwsAuthenticatorCommandArgs(): string[] | undefined {
    return this.inputs['kubeconfig_aws_authenticator_command_args'] as string[] | undefined;
  }
  public set kubeconfigAwsAuthenticatorCommandArgs(value: string[] | undefined) {
    this.inputs['kubeconfig_aws_authenticator_command_args'] = value;
  }
  public get kubeconfigAwsAuthenticatorEnvVariables(): { [key: string]: string } | undefined {
    return this.inputs['kubeconfig_aws_authenticator_env_variables'] as { [key: string]: string } | undefined;
  }
  public set kubeconfigAwsAuthenticatorEnvVariables(value: { [key: string]: string } | undefined) {
    this.inputs['kubeconfig_aws_authenticator_env_variables'] = value;
  }
  public get kubeconfigName(): string | undefined {
    return this.inputs['kubeconfig_name'] as string | undefined;
  }
  public set kubeconfigName(value: string | undefined) {
    this.inputs['kubeconfig_name'] = value;
  }
  public get localExecInterpreter(): string[] | undefined {
    return this.inputs['local_exec_interpreter'] as string[] | undefined;
  }
  public set localExecInterpreter(value: string[] | undefined) {
    this.inputs['local_exec_interpreter'] = value;
  }
  public get manageAwsAuth(): boolean | undefined {
    return this.inputs['manage_aws_auth'] as boolean | undefined;
  }
  public set manageAwsAuth(value: boolean | undefined) {
    this.inputs['manage_aws_auth'] = value;
  }
  public get manageClusterIamResources(): boolean | undefined {
    return this.inputs['manage_cluster_iam_resources'] as boolean | undefined;
  }
  public set manageClusterIamResources(value: boolean | undefined) {
    this.inputs['manage_cluster_iam_resources'] = value;
  }
  public get manageWorkerAutoscalingPolicy(): boolean | undefined {
    return this.inputs['manage_worker_autoscaling_policy'] as boolean | undefined;
  }
  public set manageWorkerAutoscalingPolicy(value: boolean | undefined) {
    this.inputs['manage_worker_autoscaling_policy'] = value;
  }
  public get manageWorkerIamResources(): boolean | undefined {
    return this.inputs['manage_worker_iam_resources'] as boolean | undefined;
  }
  public set manageWorkerIamResources(value: boolean | undefined) {
    this.inputs['manage_worker_iam_resources'] = value;
  }
  public get mapAccounts(): string[] | undefined {
    return this.inputs['map_accounts'] as string[] | undefined;
  }
  public set mapAccounts(value: string[] | undefined) {
    this.inputs['map_accounts'] = value;
  }
  public get mapRoles(): any | undefined {
    return this.inputs['map_roles'] as any | undefined;
  }
  public set mapRoles(value: any | undefined) {
    this.inputs['map_roles'] = value;
  }
  public get mapUsers(): any | undefined {
    return this.inputs['map_users'] as any | undefined;
  }
  public set mapUsers(value: any | undefined) {
    this.inputs['map_users'] = value;
  }
  public get permissionsBoundary(): string | undefined {
    return this.inputs['permissions_boundary'] as string | undefined;
  }
  public set permissionsBoundary(value: string | undefined) {
    this.inputs['permissions_boundary'] = value;
  }
  public get subnets(): string[] {
    return this.inputs['subnets'] as string[];
  }
  public set subnets(value: string[]) {
    this.inputs['subnets'] = value;
  }
  public get tags(): { [key: string]: string } | undefined {
    return this.inputs['tags'] as { [key: string]: string } | undefined;
  }
  public set tags(value: { [key: string]: string } | undefined) {
    this.inputs['tags'] = value;
  }
  public get vpcId(): string {
    return this.inputs['vpc_id'] as string;
  }
  public set vpcId(value: string) {
    this.inputs['vpc_id'] = value;
  }
  public get workerAdditionalSecurityGroupIds(): string[] | undefined {
    return this.inputs['worker_additional_security_group_ids'] as string[] | undefined;
  }
  public set workerAdditionalSecurityGroupIds(value: string[] | undefined) {
    this.inputs['worker_additional_security_group_ids'] = value;
  }
  public get workerAmiNameFilter(): string | undefined {
    return this.inputs['worker_ami_name_filter'] as string | undefined;
  }
  public set workerAmiNameFilter(value: string | undefined) {
    this.inputs['worker_ami_name_filter'] = value;
  }
  public get workerAmiNameFilterWindows(): string | undefined {
    return this.inputs['worker_ami_name_filter_windows'] as string | undefined;
  }
  public set workerAmiNameFilterWindows(value: string | undefined) {
    this.inputs['worker_ami_name_filter_windows'] = value;
  }
  public get workerAmiOwnerId(): string | undefined {
    return this.inputs['worker_ami_owner_id'] as string | undefined;
  }
  public set workerAmiOwnerId(value: string | undefined) {
    this.inputs['worker_ami_owner_id'] = value;
  }
  public get workerAmiOwnerIdWindows(): string | undefined {
    return this.inputs['worker_ami_owner_id_windows'] as string | undefined;
  }
  public set workerAmiOwnerIdWindows(value: string | undefined) {
    this.inputs['worker_ami_owner_id_windows'] = value;
  }
  public get workerCreateInitialLifecycleHooks(): boolean | undefined {
    return this.inputs['worker_create_initial_lifecycle_hooks'] as boolean | undefined;
  }
  public set workerCreateInitialLifecycleHooks(value: boolean | undefined) {
    this.inputs['worker_create_initial_lifecycle_hooks'] = value;
  }
  public get workerCreateSecurityGroup(): boolean | undefined {
    return this.inputs['worker_create_security_group'] as boolean | undefined;
  }
  public set workerCreateSecurityGroup(value: boolean | undefined) {
    this.inputs['worker_create_security_group'] = value;
  }
  public get workerGroups(): any | undefined {
    return this.inputs['worker_groups'] as any | undefined;
  }
  public set workerGroups(value: any | undefined) {
    this.inputs['worker_groups'] = value;
  }
  public get workerGroupsLaunchTemplate(): any | undefined {
    return this.inputs['worker_groups_launch_template'] as any | undefined;
  }
  public set workerGroupsLaunchTemplate(value: any | undefined) {
    this.inputs['worker_groups_launch_template'] = value;
  }
  public get workerSecurityGroupId(): string | undefined {
    return this.inputs['worker_security_group_id'] as string | undefined;
  }
  public set workerSecurityGroupId(value: string | undefined) {
    this.inputs['worker_security_group_id'] = value;
  }
  public get workerSgIngressFromPort(): number | undefined {
    return this.inputs['worker_sg_ingress_from_port'] as number | undefined;
  }
  public set workerSgIngressFromPort(value: number | undefined) {
    this.inputs['worker_sg_ingress_from_port'] = value;
  }
  public get workersAdditionalPolicies(): string[] | undefined {
    return this.inputs['workers_additional_policies'] as string[] | undefined;
  }
  public set workersAdditionalPolicies(value: string[] | undefined) {
    this.inputs['workers_additional_policies'] = value;
  }
  public get workersGroupDefaults(): any | undefined {
    return this.inputs['workers_group_defaults'] as any | undefined;
  }
  public set workersGroupDefaults(value: any | undefined) {
    this.inputs['workers_group_defaults'] = value;
  }
  public get workersRoleName(): string | undefined {
    return this.inputs['workers_role_name'] as string | undefined;
  }
  public set workersRoleName(value: string | undefined) {
    this.inputs['workers_role_name'] = value;
  }
  public get writeAwsAuthConfig(): boolean | undefined {
    return this.inputs['write_aws_auth_config'] as boolean | undefined;
  }
  public set writeAwsAuthConfig(value: boolean | undefined) {
    this.inputs['write_aws_auth_config'] = value;
  }
  public get writeKubeconfig(): boolean | undefined {
    return this.inputs['write_kubeconfig'] as boolean | undefined;
  }
  public set writeKubeconfig(value: boolean | undefined) {
    this.inputs['write_kubeconfig'] = value;
  }
  public get cloudwatchLogGroupNameOutput(): string {
    return this.interpolationForOutput('cloudwatch_log_group_name')
  }
  public get clusterArnOutput(): string {
    return this.interpolationForOutput('cluster_arn')
  }
  public get clusterCertificateAuthorityDataOutput(): string {
    return this.interpolationForOutput('cluster_certificate_authority_data')
  }
  public get clusterEndpointOutput(): string {
    return this.interpolationForOutput('cluster_endpoint')
  }
  public get clusterIamRoleArnOutput(): string {
    return this.interpolationForOutput('cluster_iam_role_arn')
  }
  public get clusterIamRoleNameOutput(): string {
    return this.interpolationForOutput('cluster_iam_role_name')
  }
  public get clusterIdOutput(): string {
    return this.interpolationForOutput('cluster_id')
  }
  public get clusterOidcIssuerUrlOutput(): string {
    return this.interpolationForOutput('cluster_oidc_issuer_url')
  }
  public get clusterSecurityGroupIdOutput(): string {
    return this.interpolationForOutput('cluster_security_group_id')
  }
  public get clusterVersionOutput(): string {
    return this.interpolationForOutput('cluster_version')
  }
  public get configMapAwsAuthOutput(): string {
    return this.interpolationForOutput('config_map_aws_auth')
  }
  public get kubeconfigOutput(): string {
    return this.interpolationForOutput('kubeconfig')
  }
  public get kubeconfigFilenameOutput(): string {
    return this.interpolationForOutput('kubeconfig_filename')
  }
  public get workerAutoscalingPolicyArnOutput(): string {
    return this.interpolationForOutput('worker_autoscaling_policy_arn')
  }
  public get workerAutoscalingPolicyNameOutput(): string {
    return this.interpolationForOutput('worker_autoscaling_policy_name')
  }
  public get workerIamInstanceProfileArnsOutput(): string {
    return this.interpolationForOutput('worker_iam_instance_profile_arns')
  }
  public get workerIamInstanceProfileNamesOutput(): string {
    return this.interpolationForOutput('worker_iam_instance_profile_names')
  }
  public get workerIamRoleArnOutput(): string {
    return this.interpolationForOutput('worker_iam_role_arn')
  }
  public get workerIamRoleNameOutput(): string {
    return this.interpolationForOutput('worker_iam_role_name')
  }
  public get workerSecurityGroupIdOutput(): string {
    return this.interpolationForOutput('worker_security_group_id')
  }
  public get workersAsgArnsOutput(): string {
    return this.interpolationForOutput('workers_asg_arns')
  }
  public get workersAsgNamesOutput(): string {
    return this.interpolationForOutput('workers_asg_names')
  }
  public get workersDefaultAmiIdOutput(): string {
    return this.interpolationForOutput('workers_default_ami_id')
  }
  public get workersLaunchTemplateArnsOutput(): string {
    return this.interpolationForOutput('workers_launch_template_arns')
  }
  public get workersLaunchTemplateIdsOutput(): string {
    return this.interpolationForOutput('workers_launch_template_ids')
  }
  public get workersLaunchTemplateLatestVersionsOutput(): string {
    return this.interpolationForOutput('workers_launch_template_latest_versions')
  }
  public get workersUserDataOutput(): string {
    return this.interpolationForOutput('workers_user_data')
  }
  protected synthesizeAttributes() {
    return this.inputs;
  }
}
"
`;

exports[`no module outputs 1`] = `
"// generated by cdktf get
// ./module
import { TerraformModule } from 'cdktf';
import { Construct } from 'constructs';
export interface ModuleOptions {
  /**
   * Flag to optionally disable usage of this module.
   * @default true
   */
  readonly enabled?: boolean;
}
export class Module extends TerraformModule {
  private readonly inputs: { [name: string]: any } = { }
  public constructor(scope: Construct, id: string, options: ModuleOptions = {}) {
    super(scope, id, {
      source: './module',
    });
    this.enabled = options.enabled;
  }
  public get enabled(): boolean | undefined {
    return this.inputs['enabled'] as boolean | undefined;
  }
  public set enabled(value: boolean | undefined) {
    this.inputs['enabled'] = value;
  }
  protected synthesizeAttributes() {
    return this.inputs;
  }
}
"
`;

exports[`typeless variables 1`] = `
"// generated by cdktf get
// ./module
import { TerraformModule } from 'cdktf';
import { Construct } from 'constructs';
export interface ModuleOptions {
  /**
   * My example var without type set, but with default
   * @default 1
   */
  readonly myDefaultTypevar?: number;
  /**
   * My example var without type set
   */
  readonly myTypelessVar: any;
}
export class Module extends TerraformModule {
  private readonly inputs: { [name: string]: any } = { }
  public constructor(scope: Construct, id: string, options: ModuleOptions) {
    super(scope, id, {
      source: './module',
    });
    this.myDefaultTypevar = options.myDefaultTypevar;
    this.myTypelessVar = options.myTypelessVar;
  }
  public get myDefaultTypevar(): number | undefined {
    return this.inputs['my_default_typevar'] as number | undefined;
  }
  public set myDefaultTypevar(value: number | undefined) {
    this.inputs['my_default_typevar'] = value;
  }
  public get myTypelessVar(): any {
    return this.inputs['my_typeless_var'] as any;
  }
  public set myTypelessVar(value: any) {
    this.inputs['my_typeless_var'] = value;
  }
  protected synthesizeAttributes() {
    return this.inputs;
  }
}
"
`;

exports[`typeless variables 2`] = `
"// generated by cdktf get
// ./module
import { TerraformModule } from 'cdktf';
import { Construct } from 'constructs';
export interface ModuleOptions {
  /**
   * My example var without type set, but with default
   * @default 1
   */
  readonly myDefaultTypevar?: number;
  /**
   * My example var without type set
   */
  readonly myTypelessVar: any;
}
export class Module extends TerraformModule {
  private readonly inputs: { [name: string]: any } = { }
  public constructor(scope: Construct, id: string, options: ModuleOptions) {
    super(scope, id, {
      source: './module',
    });
    this.myDefaultTypevar = options.myDefaultTypevar;
    this.myTypelessVar = options.myTypelessVar;
  }
  public get myDefaultTypevar(): number | undefined {
    return this.inputs['my_default_typevar'] as number | undefined;
  }
  public set myDefaultTypevar(value: number | undefined) {
    this.inputs['my_default_typevar'] = value;
  }
  public get myTypelessVar(): any {
    return this.inputs['my_typeless_var'] as any;
  }
  public set myTypelessVar(value: any) {
    this.inputs['my_typeless_var'] = value;
  }
  protected synthesizeAttributes() {
    return this.inputs;
  }
}
"
`;
